### PR TITLE
[NAE-1623] Action API consolidation

### PR DESCRIPTION
**Pull request by petrovicLubos**
# Description

Implemented overloaded function for PDF generator in ActionDelegate.

Implements [NAE-1623]

## Dependencies

No new dependencies were introduced.

### Third party dependencies

No new dependencies were introduced.

### Blocking Pull requests

There are no dependencies on other PR.

## How Has Been This Tested?

This was tested manually using developed functions. All tests went very well.

### Test Configuration

Netgrif Backend PR Config

| Name                | Tested on |
|---------------------| --------- |
| OS                  |   windows 10 OS        |
| Runtime             |  Java 11        |
| Dependency Manager  |  Maven 3.8.5         |
| Framework version   |  Spring Boot 2.6.2        |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x]  My changes have been checked, personally or remotely, with @mladoniczky 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1623]: https://netgrif.atlassian.net/browse/NAE-1623

### CODE REVIEW

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy [hunk 1]
```diff
@@ -244,7 +248,17 @@ class ActionDelegate {
                     this.outcomes.add(outcome)
                 }
             }]
-        }]
+        },
+         on: { Transition trans ->
+             behavior(field, trans)
```
Comment (minop) [inline src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:253]: duplicate code with the above implementation

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy [hunk 2]
```diff
@@ -787,6 +812,17 @@ class ActionDelegate {
         change useCase.getField(fileFieldId) value { new FileFieldValue(filename, storagePath) }
     }
 
+    void generatePDF(String transitionId, FileField fileField) {
```
Comment (minop) [inline src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:815]: how does the introduction of these methods relate to the changes made to these methods in PRs #39 and #40 ?

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy [hunk 3]
```diff
@@ -953,6 +1010,26 @@ class ActionDelegate {
...
+            return null
+        } else {
+            return user
+        }
+    }
+
+    IUser findUserById(String id) {
+        IUser user = userService.findById(id, false)
+        if (user == null){
+            log.error("Cannot find user with email [" + email + "]")
```
Comment (minop) [inline src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:1026]: the error message does not correspond to the implementation

**Review by minop — CHANGES_REQUESTED** [review]

**Comment by machacjozef** [comment]
### Please edit:

- [ ] Title
- [ ] Description
- [ ] Change commit message
- [ ] Fix Test

**Review by machacjozef — CHANGES_REQUESTED** [review]
### Please make changes:

- [ ] edit Title
- [ ] edit Description
- [ ] Change commit message
- [x] Fix Test

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy [hunk 4]
```diff
@@ -244,7 +248,17 @@ class ActionDelegate {
                     this.outcomes.add(outcome)
                 }
             }]
-        }]
+        },
+         on: { Transition trans ->
```
Comment (RudeBoyxX) [inline src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:252]: make method improvements are part of https://netgrif.atlassian.net/browse/NAE-1622 (currently in progress)

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy (hunk 2 above)
Comment (minop) [reply src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:815]: @renczesstefan informed me that he will resolve conflicted changes made to these methods once the related PRs are ready to merge

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy [hunk 5]
```diff
@@ -244,7 +248,7 @@ class ActionDelegate {
                     this.outcomes.add(outcome)
                 }
             }]
-        }]
```
Comment (minop) [inline src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:247]: the `]` bracket is missing

**Review by minop — CHANGES_REQUESTED** [review]

**Review by minop — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Review by machacjozef — APPROVED** [review]

File: src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy [hunk 6]
```diff
@@ -779,17 +794,32 @@ class ActionDelegate {
...
+        pdfResource.setMarginTitle(titleMargin)
+        pdfResource.setMarginLeft(sideMargin)
+        pdfResource.setMarginRight(sideMargin)
+        pdfResource.updateProperties()
         pdfGenerator.setupPdfGenerator(pdfResource)
         pdfGenerator.generatePdf(useCase, transitionId, pdfResource)
         change useCase.getField(fileFieldId) value { new FileFieldValue(filename, storagePath) }
     }
 
+    void generatePDF(String transitionId, FileField fileField) {
```
Comment (minop) [inline src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy:812]: this method should probably call the `generatePDF(String, FileField, List<String>)` method or make use of groovy default parameter values to reduce duplication

**Review by minop — CHANGES_REQUESTED** [review]

**Review by minop — DISMISSED** [review]

**Review by machacjozef — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Review by minop — DISMISSED** [review]

**Review by mladoniczky — APPROVED** [review]

**Review by machacjozef — APPROVED** [review]

**Pull request merged**